README.md

```diff
@@ -15,24 +15,169 @@ Rule::allClasses()
+    ->because('it configures the service container');
+```
+
+### Have a name matching a pattern
+
+```
+$rules = Rule::allClasses()
+    ->that(new ResideInOneOfTheseNamespaces('App\Service'))
+    ->should(new HaveNameMatching('*Service'))
+    ->because('we want uniform naming for services');
+```
+
+### Implements an interface
+
+```
+$rules = Rule::allClasses()
+    ->that(new ResideInOneOfTheseNamespaces('App\Controller'))
+    ->should(new Implement('ContainerAwareInterface'))
+    ->because('all controllers should be container aware');
+```
+
+### Not implements an interface
+
+```
+$rules = Rule::allClasses()
+    ->that(new ResideInOneOfTheseNamespaces('App\Infrastructure\RestApi\Public'))
+    ->should(new NotImplement('ContainerAwareInterface'))
+    ->because('all public controllers should not be container aware');
+```
+
+### Is abstract
+
+```
+$rules = Rule::allClasses()
+    ->that(new ResideInOneOfTheseNamespaces('App\Customer\Service'))
+    ->should(new IsAbstract())
+    ->because('we want to be sure that classes are abstract in a specific namespace');
+```
+
+### Is final
+
+```
+$rules = Rule::allClasses()
+    ->that(new ResideInOneOfTheseNamespaces('App\Domain\Aggregates'))
+    ->should(new IsFinal())
+    ->because('we want to be sure that aggregates are final classes');
+```
+
+### Is not abstract
+
+```
+$rules = Rule::allClasses()
+    ->that(new ResideInOneOfTheseNamespaces('App\Domain'))
+    ->should(new IsNotAbstract())
+    ->because('we want to avoid abstract classes into our domain');
+```
+
+### Is not final
+
+```
+$rules = Rule::allClasses()
+    ->that(new ResideInOneOfTheseNamespaces('App\Infrastructure\Doctrine'))
+    ->should(new IsNotFinal())
+    ->because('we want to be sure that our adapters are not final classes');
+```
+
+### Not depends on a namespace
+
+```
+$rules = Rule::allClasses()
+    ->that(new ResideInOneOfTheseNamespaces('App\Application'))
+    ->should(new NotDependsOnTheseNamespaces('App\Infrastructure'))
+    ->because('we want to avoid coupling between application layer and infrastructure layer');
+```
+
+### Not extend another class
+
+```
+$rules = Rule::allClasses()
+    ->that(new ResideInOneOfTheseNamespaces('App\Controller\Admin'))
+    ->should(new NotExtend('App\Controller\AbstractController'))
+    ->because('we want to be sure that all admin controllers not extend AbstractController for security reasons');
+```
+
+### Don't have dependency outside a namespace
+
+```
+$rules = Rule::allClasses()
+    ->that(new ResideInOneOfTheseNamespaces('App\Domain'))
+    ->should(new NotHaveDependencyOutsideNamespace('App\Domain', ['Ramsey\Uuid']))
+    ->because('we want protect our domain except for Ramsey\Uuid');
+```
+
+### Not have a name matching a pattern
+
+```
+$rules = Rule::allClasses()
+    ->that(new ResideInOneOfTheseNamespaces('App'))
+    ->should(new NotHaveNameMatching('*Manager'))
+    ->because('we want to avoid using Manager for our classes');
```

Comment:
Maybe something more like:
"*Manager is too vague in naming classes"

Reply:
I agree, much better.